js/view.js

```diff
@@ -0,0 +1,132 @@
+goog.module('wordsearch.view');
+
+const model = goog.require('wordsearch.model');
+
+const canvas = document.getElementById('wordSearchCanvas');
+const context = canvas.getContext('2d');
+
+/**
+ * Returns the height of the view
+ * @returns {number}
+ */
+function getViewHeight() {
+  return canvas.getBoundingClientRect().height;
+}
+
+/**
+ * Returns the width of the view
+ * @returns {number}
+ */
+function getViewWidth() {
+  return canvas.getBoundingClientRect().width;
+}
+
+class View {
+  /** @param {!model.GameBoard} wordSearchBoard */
+  constructor(wordSearchBoard) {
+    /** @const {!model.GameBoard} */
+    this.wordSearchBoard = wordSearchBoard;
+  }
+
+  /*
+   * Draws the game board onto the screen with the canvas api.
+   */
+  drawGameBoard() {
+    context.textAlign = 'center';
+    context.font = '30px Arial';
+    for (const row of this.wordSearchBoard.board) {
+      for (const square of row) {
+        context.fillStyle = square.colour;
+        context.fillRect(
+            square.xCoord, square.yCoord,
+            this.wordSearchBoard.dimensionsOfSquare,
+            this.wordSearchBoard.dimensionsOfSquare);
+
+        context.fillStyle = 'black';
+        context.fillText(
+            square.character,
+            square.xCoord + this.wordSearchBoard.dimensionsOfSquare / 2,
+            square.yCoord + this.wordSearchBoard.dimensionsOfSquare / 2 + 10);
+      }
+    }
+  }
+
+  /**
+   * Draws a line to indicate the current selection within the game board
+   * between the two points
+   * @param {!model.Point} startPoint
+   * @param {!model.Point} tempEndPoint
+   */
+  drawSearchLine(startPoint, tempEndPoint) {
+    const angleRadians = Math.atan2(
+        tempEndPoint.y - startPoint.y, tempEndPoint.x - startPoint.x);
+
+    const dist = Math.sqrt(
+        Math.pow(startPoint.x - tempEndPoint.x, 2) +
+        Math.pow(startPoint.y - tempEndPoint.y, 2))
+
+    // We re-adjust the endPoint so that both points lie on a horizantol line.
+    // This makes the math of determining the midpoint easier, and allows us to
+    // rotate the rectangle with respect to the real endpoint.
+    const endPoint = new model.Point(startPoint.x + dist, startPoint.y);
+
+    const radiusOfLine = 15;
+
+    const midPoint = startPoint.x + ((endPoint.x - startPoint.x) / 2);
+
+    context.clearRect(0, 0, canvas.width, canvas.height);
+
+    this.drawGameBoard();
```

Comment:
Hm. Wondering if it's possible to redraw board quickly without iterating through all squares. One way could be having 2 canvases, one is visible and one is invisible. The invisible one contains drawn board without lines. And here you'll just copy content of invisible canvas to the visible. Maybe it's faster. 

But I don't think it's important yet. When/if you have free time you can try increasing number of letters to 1000 or more and see how it performs.

Reply:
Added a TODO for now.